pkg/cluster/majorversionupgrade.go

```diff
@@ -55,6 +56,27 @@ func (c *Cluster) isUpgradeAllowedForTeam(owningTeam string) bool {
 	return util.SliceContains(allowedTeams, owningTeam)
 }
 
+func (c *Cluster) isInMainternanceWindow() bool {
```

Comment:
can we also have unit test for this function? 🥺  (e.g. ,the input scenarios you've tested manually)